### PR TITLE
Upgrade digitalbuildings tools to depend on Python 3.11

### PR DESCRIPTION
**Pull request by trav3711**
Upgrading python version for all tools to be >= Python 3.11

### CODE REVIEW

**Comment by trav3711** [comment]
I can find the bug that keeps breaking the scoring tool. It's not any of the parsing logic because the scoring tool depends on the instance validator's `Deserilaize` method for that which passes its own tests. My hunch is something in python's standard library has changed causing the scoring tool to break with the 3.11 migration. 

I was thinking the `map()` method changed but no dice yet. There's also not much documentation for python 3.11 bugs yet, but the release is considered stable.

https://www.geeksforgeeks.org/python-map-function/

**Comment by tasodorff** [comment]
@trav3711 is this still valid or should we close?

**Comment by sypks** [comment]
patched scoring tool and running github actions now

**Comment by sypks** [comment]
patch success. please review @trav3711 @tasodorff @charbull

File: tools/scoring/score/parse_config.py [hunk 1]
```diff
@@ -181,8 +181,8 @@ def aggregate_translations(entity) -> List[Any]:
         return []
 
       translations[cloud_device_id] = {
-          f'{PROPOSED}': aggregate_translations(proposed_entity),
-          f'{SOLUTION}': aggregate_translations(solution_entity)
+          PROPOSED: aggregate_translations(proposed_entity),
```
Comment (trav3711) [inline tools/scoring/score/parse_config.py:184]: Why did the f string literal break the scoring tool? Is that a change in 3.11?

**Comment by trav3711** [comment]
Looks like you need to get your CLA approved? Not sure how this is done but I thought you did this already?

File: tools/abel/setup.py [hunk 2]
```diff
@@ -26,5 +26,5 @@
     install_requires=[
         'absl-py', 'pyfiglet', 'argparse', 'pyyaml', 'google-api-python-client'
     ],
-    python_requires='>=3.9',
+    python_requires='>=3.11',
```
Comment (charbull) [inline tools/abel/setup.py:29]: I propose to keep it = for now so we dont encounter future changes/surprises, wdyt?

**Review by charbull — CHANGES_REQUESTED** [review]
make the version stricter

**Pull request closed without merging**